win32ss/user/ntuser/sysparams.c

```diff
@@ -942,18 +1022,57 @@ SpiGetSet(UINT uiAction, UINT uiParam, PVOID pvParam, FLONG fl)
 
         case SPI_GETNONCLIENTMETRICS:
         {
-            return SpiGet(pvParam, &gspv.ncm, sizeof(NONCLIENTMETRICSW), fl);
+            if (uiParam == 0 && !SpiGetInt(pvParam, &uiParam, fl))
+                return 0;
+
+            if (uiParam == sizeof(NONCLIENTMETRICSA))
```

Comment:
Regarding that size check: I'm not sure this constraint (passing a redundant `sizeof(NONCLIENTMETRICS)` as parameter of `SystemParametersInfo` is systematically verified, on Windows 2003 (which is the version we are being compatible with).
This definitively requires an API test.